draft-ietf-quic-transport.md

```diff
@@ -1618,69 +1618,84 @@ of three ways:
+This state runs for the same duration as the draining period.  While in the
+closing period, an endpoint MUST NOT send packets unless they contain a
+CONNECTION_CLOSE or APPLICATION_CLOSE frame (see {{immediate-close}} for
+details).
 
-The draining period does not apply when a stateless reset ({{stateless-reset}})
-is sent.
+The closing period differs from the draining period only in that closing frames
+can be retransmitted.
+
+An endpoint MAY transition from the closing period to the draining period if it
+can confirm that its peer is also closing or draining.  Receiving a closing
+frame is sufficient confirmation, as is receiving a stateless reset.  The
+draining period SHOULD end when the closing period would have ended.  In other
+words, the endpoint can use the same end time, but cease retransmission of the
+closing packet.
+
+The draining and closing periods do not apply when a stateless reset
+({{stateless-reset}}) is sent.
 
 
 ### Idle Timeout
 
 A connection that remains idle for longer than the idle timeout (see
-{{transport-parameter-definitions}}) becomes closed.  Either peer removes
-connection state if they have neither sent nor received a packet for this time.
+{{transport-parameter-definitions}}) is closed.  At implementation discretion, a
+connection enters either the draining or closing period when the idle timeout
+expires.
 
 The time at which an idle timeout takes effect won't be perfectly synchronized
-on peers.  A connection enters the draining period when the idle timeout
-expires.  During this time, an endpoint that receives new packets MAY choose to
-restore the connection.  Alternatively, an endpoint that receives packets MAY
-signal the timeout using an immediate close.
+on peers.  An endpoint MAY enter a third state where receipt of new packets
+causes the connection to be restored to normal operation.
 
 
 ### Immediate Close
 
-An endpoint sends a CONNECTION_CLOSE or APPLICATION_CLOSE frame to terminate the
-connection immediately.  Either frame causes all open streams to immediately
-become closed; open streams can be assumed to be implicitly reset.  After
-sending or receiving a CONNECTION_CLOSE frame, endpoints immediately enter a
-draining period.
-
-During the draining period, an endpoint that sends a CONNECTION_CLOSE or
-APPLICATION_CLOSE frame SHOULD respond to any subsequent packet that it receives
-with another packet containing either close frame.  To reduce the state that an
-endpoint maintains in this case, it MAY send the exact same packet.  However,
-endpoints SHOULD limit the number of packets they generate containing either
-close frame.  For instance, an endpoint could progressively increase the number
-of packets that it receives before sending additional packets.
+An endpoint sends a closing frame, either CONNECTION_CLOSE or APPLICATION_CLOSE,
+to terminate the connection immediately.  Either closing frame causes all open
```

Comment:
Nit:  You don't need "open" here.  All streams are immediately closed; open streams get reset.